.github/workflows/swift-toolchain.yml

```diff
@@ -930,55 +934,94 @@ jobs:
         uses: hendrikmuhs/ccache-action@2e0e89e8d74340a03f75d58d02aae4c5ee1b15c6
         with:
           max-size: 500M
-          key: ${{ steps.workspace_hash.outputs.hash }}-windows-${{ matrix.arch }}-compilers
+          key: ${{ steps.workspace_hash.outputs.hash }}-${{ matrix.os }}-${{ matrix.arch }}-compilers
           variant: sccache
 
       - name: Configure Compilers
         env:
           NDKPATH: ${{ steps.setup-ndk.outputs.ndk-path }}
         run: |
-          if ( "${{ matrix.arch }}" -eq "arm64" ) {
-            $CMAKE_SYSTEM_NAME="-D CMAKE_SYSTEM_NAME=Windows"
-            $CMAKE_SYSTEM_PROCESSOR="-D CMAKE_SYSTEM_PROCESSOR=ARM64"
-            $CACHE="Windows-aarch64.cmake"
-
-            # FIXME(compnerd) re-enable runtimes after we sort out compiler-rt
-            (Get-Content ${{ github.workspace }}/SourceCache/swift/cmake/caches/Windows-aarch64.cmake).Replace(' runtimes', '') | Set-Content ${{ github.workspace }}/SourceCache/swift/cmake/caches/Windows-aarch64.cmake
-          } else {
-            $CACHE="Windows-x86_64.cmake"
+          $CxxFlags = "${{ matrix.cxxflags }}"
+          $SwiftFlags = ""
+
+          if ( "${{ matrix.os }}" -eq "Windows" ) {
+            $SWIFTC = cygpath -m (Get-Command swiftc).Source
+            # Use toolchain clang to avoid broken __prefetch intrinsic on arm64 in Clang 18.
+            # TODO: Use llvm-19 when available. See https://github.com/compnerd/swift-build/issues/846
+            $CLANG_LOCATION = Split-Path (Get-Command swiftc).Source
+            $EXTRA_FLAGS = @("-D", "SWIFT_NATIVE_SWIFT_TOOLS_PATH=${{ github.workspace }}/BinaryCache/0/bin")
+            if ( "${{ matrix.arch }}" -eq "arm64" ) {
+              $EXTRA_FLAGS += "${{ matrix.extra_flags }}".Split()
+              $EXTRA_FLAGS += @("-D", "CMAKE_SYSTEM_NAME=Windows")
+              $CACHE="${{ github.workspace }}/SourceCache/swift/cmake/caches/Windows-aarch64.cmake"
+
+              # FIXME(compnerd) re-enable runtimes after we sort out compiler-rt
```

Comment:
I think that we should check with @weliveindetail about this since he was fixing this.

Reply:
I encountered build from setting `CMAKE_SYSTEM_PROCESSOR` and/or `CMAKE_SYSTEM_NAME` - hence why they are disabled in the x64 build. Since we do need to at least set `CMAKE_SYSTEM_PROCESSOR` when cross-compiling for arm64, this might be the same issue.